StRoot/StarGenerator/BASE/AgStarReader.h

```diff
@@ -22,6 +22,10 @@ class AgStarReader : public TObject
   /// @param stack The particle stack
   void SetStack( StarParticleStack *stack ){ mStack = stack; }
 
+  /// Set the particle data
+  /// @param data Pointer to the particle data
+  void SetParticleData( StarParticleData* data ){ mParticleData = data; }
```

Comment:
Why add this setter and keep the field if you replace it with a call to `StarParticleData::instance()` in the `AgStarReader::ReadEvent()`?

Reply:
Solution to a problem that I had to go a different direction on.  Removed.